### ISSUE TITLE
No way to customize lock timeout

### ISSUE THREAD

**Issue by JoostHouben** [fruux/sabre-dav#876]
When using the `Locks` plugin with either the `PDO` or `File` backend, the lock timeout is always set to 30 minutes. Modifying this behaviour seems like a common use case (for example: use a different default, or actually honour the clients request). But at the moment this requires copy pasting the entire `lock` method just to modify the one statement that sets the lock timeout. This method should be split up into smaller submethods so that inheritors can more easily customize behaviour. Or, at the very least, there should be some way to configure the timeout that is imposed by the backend.

**Comment by evert** [comment]
Hi Joost,

I originally set the timeout to 30 minutes thinking: if somebody needs a different behavior, I'll probably get a feature request. 8 years later, here you are. Welcome!

However, as you might have deduced, it's not a very common use-case. Can you elaborate on your use-case? Is it hypothetical or is there a more specific reason you want this?

**Comment by JoostHouben** [comment]
Yes, there is a specific reason: Microsoft Office doesn't always send a valid unlock request. In particular, if authentication is used, MS Office sometimes sends an unlock request without an authorization header. To reduce the impact of hanging locks, I reduced the length of the timeout.

**Comment by evert** [comment]
Good enough for me!
